New package: KaitaiStruct v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: KaitaiStruct
- Repository: https://github.com/rystidia/KaitaiStruct.jl
- Created by: @rystidia
- Version: v0.1.0
- Commit: 92ba99b5c22febafdc6fe2618ab11a34c0fcdda8
- Reviewed by: @rystidia
- Reference: https://github.com/rystidia/KaitaiStruct.jl/commit/92ba99b5c22febafdc6fe2618ab11a34c0fcdda8#commitcomment-141353862
- Description: Kaitai Struct: runtime for Julia
- Release notes:
<!-- BEGIN RELEASE NOTES -->
`````
the first version of Julia runtime library for Kaitai Struct
`````
<!-- END RELEASE NOTES -->

<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd6b751678f94ccbe1d6016b60f5afe25fdb2d4ffca47513e63ae5e307ea05f39f66245613bbedf97a901a93e4daa63a186e234326f6e252c6cb403d45bbf2b3546fe7dd9a8b23c46e4ea639ae2d70ce095b1057fa1817fe0d7f6fa13aa05200b127efdb281bf72ed4dd0889621c25bc1b6a89e6e6ecad81ef5f9004d290b967daa53fe1c98351520dcc181ae9c8153079ab8ba5c1d025de6ed04f3033e0276a78196e02bc5731ea93cc0e84653583b433b -->

rystidia commented:
@goerz 
It is meant to be used by generated parsing modules, not users. [Kaitai Struct ](https://kaitai.io/) is a parser generator. It offers a declarative domain-specific language KSY to describe the structure of binary data. Format descriptions in that language can be transpiled to parsers in 11 programming languages (I want to add Julia as the 12th supported language). 
For example the following specification defines a string preceded by an integer that designates its length:
```yaml
seq:
  - id: my_len
    type: u4
  - id: my_str
    type: str
    size: my_len
    encoding: UTF-8
 ```
 With this library, it can be parsed in the following way:
 ```julia
 my_len = KaitaiStruct.readU4(io)
my_str = decode(KaitaiStruct.read_bytes(io, my_len), "UTF-8")
```
However, the library is not meant to be used by a user as a standalone project.  Should I add documentation and clarify it there? Thank you.

goerz commented:
[noblock] if you put exactly that kind of information in the README, that should be totally fine. You can almost just copy and paste it.

goerz commented:
[noblock] After you fix the Julia compatibility issue, make sure to also unblock your own comment

JuliaRegistrator committed:
New package: KaitaiStruct v0.1.0

UUID: 701c8e6d-1b01-4551-886b-1760f51f9f8d
Repo: https://github.com/rystidia/KaitaiStruct.jl.git
Tree: f40b2f511d87ace91a3b92b26a1be2bd35723de7

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a
